src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AlignedAssemblyOrExcuse.java

```diff
@@ -122,21 +117,32 @@ public FermiLiteAssembly getAssembly() {
      */
     static void writeSAMFile( final String samFile,
                                      final SAMFileHeader header,
-                                     final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList ) {
-        try ( final OutputStream os = BucketUtils.createFile(samFile) ) {
-            final SAMTextWriter writer = new SAMTextWriter(os);
-            writer.setSortOrder(SAMFileHeader.SortOrder.queryname, true);
-            writer.setHeader(header);
-
+                                     final List<AlignedAssemblyOrExcuse> alignedAssemblyOrExcuseList, final boolean preOrdered ) {
+        try ( final SAMFileWriter writer = createSAMFileWriter(samFile, header, preOrdered) ) {
```

Comment:
Not sure what are you referring to... the original code and the new code always always call ``` BucketUtils.createFile(samFile)``` to instantiate the output file (check out ```createSAMFileWriter``` code). So I guess that if this new code cannot work with a HDFS destination then master has the same issue.